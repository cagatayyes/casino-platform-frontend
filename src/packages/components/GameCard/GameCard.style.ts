import tw from 'tailwind-styled-components';

export const GameCardContainer = tw.div`
  flex
  flex-col
  w-[113px]
  h-[168px]
  bg-[#070C26]
  rounded-[8px]
  last-of-type:mr-[0]
`;

export const GameCardImageContainer = tw.div`
  flex
  flex-col
  relative
  items-center
  h-[147px]
`;

export const GameCardImage = tw.img`
  flex
  rounded-[8px]
  h-full
  w-full
`;

export const GameTitleContainer = tw.span`
	absolute
	bottom-[0px]
	flex
	flex-col
	text-[#F1F1F1]
	text-center
`;

export const TitleThin = tw.div`
  flex
  font-normal
  text-[20px]
  tracking-[-0.4px]
`;

export const TitleBold = tw.div`
  flex
  font-bold
  text-[25px]
  tracking-[-0.5px]
`;

export const GameCardInfoContainer = tw.div`
  flex
  flex-row
  justify-between
  items-center
  h-[24px]
  px-[6px]
`;

export const GameInfoTitle = tw.div`
  block
  font-bold
  text-[10px]
  text-[#4C5576]
  overflow-hidden
  whitespace-nowrap
  text-ellipsis
  w-[95px]
`;

export const GameInfoAmount = tw.div`
  flex
  font-bold
  text-[11px]
  text-[#E5206F]
`;

export const GameInfoIconContainer = tw.span`
  flex
  text-[14px]
  text-[#687090]
`;
