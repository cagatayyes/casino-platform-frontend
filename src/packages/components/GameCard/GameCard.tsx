import { GameCardContainer, GameCardImage, GameCardImageContainer, GameCardInfoContainer, GameInfoAmount, GameInfoTitle } from './GameCard.style';
import { Sash } from 'casino/components/GameCard/Sash/Sash';
import { ISash } from 'casino/types';
import slugify from 'slugify';
import { ROUTES } from 'casino/constants';
import { useScreenDetector } from 'casino/hooks';

interface IGameCardProps {
	showamount: boolean;
	gamename: string;
	gameimage: string;
	gameid: number;
	sash: ISash | null;
	gamelaunchpoint: string;
	categoryName?: string;
	showgameinfomodal: (gameId: number, gameLink: string) => void;
	handlegameload: (gameLink: string, gameName: string, gamelaunchpoint: string, categoryName?: string) => void;
}

export const GameCard = ({ showamount, gamename, gameimage, sash, gameid, gamelaunchpoint, categoryName = '', /* showgameinfomodal, */ handlegameload }:
	IGameCardProps): JSX.Element => {
	const formattedGameName = slugify(`${gamename}`).toLowerCase();
	const gameLink: string = `${ROUTES.PAGES.GAME}/${formattedGameName}/${gameid}`;
	const { isSmallScreen } = useScreenDetector();

	/*
	TODO this method will be used when we enable info icon on gameCard. do not delete it
	const handleOnClickInfo = (): void => {
		showgameinfomodal?.(gameid, gameLink);
	}; */

	return (
		<GameCardContainer style={{ width: isSmallScreen ? 105 : 113 }}>
			<GameCardImageContainer onClick={() => handlegameload(gameLink, gamename, gamelaunchpoint, categoryName)}>
				{
					sash &&
					<Sash title={sash.name}
						iconName={sash.icon}
						bgColor={sash.color}
						sashStyles='right-[3px] absolute' />
				}

				<GameCardImage src={gameimage} />
			</GameCardImageContainer>
			<GameCardInfoContainer>
				{
					showamount ?
						<GameInfoAmount>
							â‚¦57,567.50 {/* TODO Amount will be different story, It will be developed in the future */}
						</GameInfoAmount>
						:
						<GameInfoTitle>
							{gamename}
						</GameInfoTitle>
				}
				{/*
				TODO this part will be used in the future do not delete it.
				<GameInfoIconContainer className='material-symbols-outlined cursor-pointer'
					onClick={handleOnClickInfo}>
					{GameCardInfoIcon}
				</GameInfoIconContainer> */}

			</GameCardInfoContainer>
		</GameCardContainer>
	);
};
