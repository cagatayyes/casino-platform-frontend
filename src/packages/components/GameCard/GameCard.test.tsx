/* eslint-disable @typescript-eslint/explicit-function-return-type */
/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import React from 'react';
import { GameCard } from './GameCard';


const sashData = {
	'sash': {
		'name': 'NEW',
		'icon': 'stars',
		'color': '#8b129b'
	}
};

jest.mock('casino/hooks/useScreenDetector', () => ({
	// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
	useScreenDetector: () => ({ isMobile: false })
}));

jest.mock('swiper/react', () => ({
	Swiper: ({ children }): JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }): JSX.Element => <div>{children}</div>
}));

jest.mock('swiper/modules', () => ({
	Swiper: ({ children }): JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }): JSX.Element => <div>{children}</div>
}));

jest.mock('react-i18next', () => ({
	useTranslation: () => ({ t: key => key }),
	Trans: ({ children }) => children
}));

describe('Game Card', () => {
	it('should render Game Card Container', () => {
		const component = render(
			<GameCard
				showamount={false}
				gamename='icecream'
				gameimage='https://casino-test.leetent.co.uk/storage//games/nSabS8jq2lplR3Wl5U7YaFHbYxp42R4YM7t89rgb.png'
				sash={sashData}
				gameid={1}
				gameLaunchPoint=''
				categoryName='lobby'
				handleGameLoad={jest.fn()}
				showgameinfomodal={jest.fn()}
			/>
		);
		expect(component).toMatchSnapshot();
	});
});
