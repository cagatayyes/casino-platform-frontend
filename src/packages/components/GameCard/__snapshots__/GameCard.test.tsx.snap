// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Card should render Game Card Container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col w-[113px] h-[168px] bg-[#070C26] rounded-[8px] last-of-type:mr-[0]"
        style="width: 113px;"
      >
        <div
          class="flex flex-col relative items-center h-[147px]"
        >
          <div
            class="flex justify-center items-center px-[4px] w-fit h-[14px] rounded-[11px] top-[3px] outline outline-[0.5px] outline-[#00000066] drop-shadow-sash right-[3px] absolute"
          >
            <span
              class="flex text-[12px] text-[#ffffff] mr-[1px] material-symbols-outlined"
            />
            <span
              class="flex text-[#ffffff] text-[10px] font-semibold"
            />
          </div>
          <img
            class="flex rounded-[8px] h-full w-full"
            src="https://casino-test.leetent.co.uk/storage//games/nSabS8jq2lplR3Wl5U7YaFHbYxp42R4YM7t89rgb.png"
          />
        </div>
        <div
          class="flex flex-row justify-between items-center h-[24px] px-[6px]"
        >
          <div
            class="block font-bold text-[10px] text-[#4C5576] overflow-hidden whitespace-nowrap text-ellipsis w-[95px]"
          >
            icecream
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col w-[113px] h-[168px] bg-[#070C26] rounded-[8px] last-of-type:mr-[0]"
      style="width: 113px;"
    >
      <div
        class="flex flex-col relative items-center h-[147px]"
      >
        <div
          class="flex justify-center items-center px-[4px] w-fit h-[14px] rounded-[11px] top-[3px] outline outline-[0.5px] outline-[#00000066] drop-shadow-sash right-[3px] absolute"
        >
          <span
            class="flex text-[12px] text-[#ffffff] mr-[1px] material-symbols-outlined"
          />
          <span
            class="flex text-[#ffffff] text-[10px] font-semibold"
          />
        </div>
        <img
          class="flex rounded-[8px] h-full w-full"
          src="https://casino-test.leetent.co.uk/storage//games/nSabS8jq2lplR3Wl5U7YaFHbYxp42R4YM7t89rgb.png"
        />
      </div>
      <div
        class="flex flex-row justify-between items-center h-[24px] px-[6px]"
      >
        <div
          class="block font-bold text-[10px] text-[#4C5576] overflow-hidden whitespace-nowrap text-ellipsis w-[95px]"
        >
          icecream
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
